pkgs/data/fonts/nerd-fonts/update.py

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -p python3 -i python3
+import os
+import urllib.request as ureq
+import json
+import subprocess
+
+RELEASE_INFO_URL = "https://api.github.com/repos/ryanoasis/nerd-fonts/releases/latest"
+FONTS_INFO_URL_TEMPLATE = "https://raw.githubusercontent.com/ryanoasis/nerd-fonts/refs/tags/{}/bin/scripts/lib/fonts.json"
+SHA256_URL_TEMPLATE = "https://github.com/ryanoasis/nerd-fonts/releases/download/{}/SHA-256.txt"
+
+RELEASE_INFO_FILENAME = "release.json"
+FONTS_INFO_FILENAME = "fonts.json"
+HASHES_FILENAME = "hashes.json"
+
+def converthash(s):
+    return subprocess.run([
+        "nix-hash",
+        "--to-sri",
+        "--type", "sha256",
+        s
+    ], capture_output=True, text=True).stdout.strip()
+
+os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), "manifests"))
+
+with ureq.urlopen(RELEASE_INFO_URL) as r:
+    release_info = json.loads(r.read())
+
+tag = release_info["tag_name"]
+with open(RELEASE_INFO_FILENAME, "r") as f:
+    former_tag = json.load(f)["tag_name"]
+if tag == former_tag:
+    raise Exception("no newer version available")
+
+ureq.urlretrieve(FONTS_INFO_URL_TEMPLATE.format(tag), FONTS_INFO_FILENAME)
+
+with ureq.urlopen(SHA256_URL_TEMPLATE.format(tag)) as r:
+    sha256 = r.read().decode("utf-8")
+hashes = [ (lambda x: {"filename": x[1], "hash": converthash(x[0])})(item.split()) for item in sha256.splitlines() ]
+with open(HASHES_FILENAME, "w") as f:
+    print(json.dumps(hashes, indent=2), file=f)
```

Comment:
This is very non Pythonic and slightly memory expensive. Try something like this:

```suggestion
hashes = {}
for line in ureq.urlopen(SHA256_URL_TEMPLATE.format(tag)):
    hash, filename = line.split()
    if filename.endswith(".zip"):
        continue
    else:
        hashes[filename] = hash
with open(HASHES_FILENAME, "w") as hf:
    json.dump(hashes, hf)
```

Slightly more functional way to do it, but still very pythonic and not memory expensive is (again, untested):

```suggestion
with open(HASHES_FILENAME, "w") as hf:
    json.dump(
        {
            filename: hash
            for line in ureq.urlopen(SHA256_URL_TEMPLATE.format(tag))
            for hash, filename in [line.split()]
            if not filename.endswith(".zip")
        },
        hf,
    )

```